### ISSUE TITLE
gnt-instance list fails when a migration is running

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#931]
Originally reported of Google Code with ID 875.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

gnt-cluster (ganeti v2.10.1) 2.10.1

Software version: 2.10.1
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.1

hspace (ganeti) version v2.10.1
compiled with ghc 7.4
running on linux x86_64


<b>What distribution are you using?</b>
  Ubuntu Precise Server

<b>What steps will reproduce the problem?</b>
1. Migrate an instance and wait for the migration to start
2. Run "gnt-instance list -o name,pnode,snodes,admin_state,oper_state,oper_ram" (simplier command probably works as well)

What is the expected output?
  An instance list.

What do you see instead?

  Unhandled protocol error while talking to the master daemon:
  Caught exception: 'migrating-<instance-name>'

<b>Please provide any additional information below.</b>

From the master-daemon.log:

2014-07-09 12:49:24,229: ganeti-masterd pid=31266/ClientReq14 WARNING Orphan instance 'migrating-<instance-name>' found on node <node-name>
2014-07-09 12:49:24,229: ganeti-masterd pid=31266/ClientReq14 ERROR Unexpected exception
Traceback (most recent call last):
  File "/usr/share/ganeti/2.10/ganeti/server/masterd.py", line 108, in RunTask
    result = client_ops.handle_request(method, args)
  File "/usr/share/ganeti/2.10/ganeti/server/masterd.py", line 350, in handle_request
    qfilter=qfilter))
  File "/usr/share/ganeti/2.10/ganeti/server/masterd.py", line 476, in _Query
    return proc.ExecOpCode(op, None)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 534, in ExecOpCode
    calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 469, in _LockAndExecLU
    result = self._LockAndExecLU(lu, level + 1, calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 469, in _LockAndExecLU
    result = self._LockAndExecLU(lu, level + 1, calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 469, in _LockAndExecLU
    result = self._LockAndExecLU(lu, level + 1, calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 469, in _LockAndExecLU
    result = self._LockAndExecLU(lu, level + 1, calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 469, in _LockAndExecLU
    result = self._LockAndExecLU(lu, level + 1, calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 469, in _LockAndExecLU
    result = self._LockAndExecLU(lu, level + 1, calc_timeout)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 408, in _LockAndExecLU
    result = self._ExecLU(lu)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 375, in _ExecLU
    result = _ProcessResult(submit_mj_fn, lu.op, lu.Exec(self.Log))
  File "/usr/share/ganeti/2.10/ganeti/cmdlib/query.py", line 85, in Exec
    return self.impl.NewStyleQuery(self)
  File "/usr/share/ganeti/2.10/ganeti/cmdlib/base.py", line 556, in NewStyleQuery
    return query.GetQueryResponse(self.query, self._GetQueryData(lu),
  File "/usr/share/ganeti/2.10/ganeti/cmdlib/instance_query.py", line 141, in _GetQueryData
    live_data[insts_by_name[iname].uuid] = result.payload[iname]
KeyError: 'migrating-<instance-name>'

```
Originally added on 2014-07-09 13:45:43 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-07-09 14:34:25 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit fb0e0879e991ea5ea525f86200d761d08fb43df8
Author: Helga Velroyen <helgav@google.com>
Date:   Thu Jul 10 09:59:33 2014 +0200

    Remove superflous loop in instance queries
    
    This patch fixes a piece of convoluted code in the
    implementation of instance queries. There was an
    unecessarily nested loop, which in case of ongoing
    migrations, could throw a key error. This fixes
    issue 875.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

```
Originally added on 2014-07-10 08:14:36 +0000 UTC.

Mentioned Issue with GoogleCode ID 875 has been migrated to GitHub with ID #931.
Added Labels: Priority-Medium
Changed State: Fixed
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.11.6
```
Originally added on 2014-09-22 15:56:40 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.11.6
```
Originally added on 2014-09-22 15:57:10 +0000 UTC.

Changed State: Released
